### PR TITLE
Fix auth on pinned triggers

### PR DESCRIPTION
**Pull request by ivov**

### CODE REVIEW

File: packages/cli/src/Server.ts [hunk 1]
```diff
@@ -3092,3 +3097,20 @@ function isOAuth(credType: ICredentialType) {
...
+	if (!pinData) return;
+
+	const firstPinnedTriggerName = Object.keys(pinData).find(isTrigger);
+
+	if (!firstPinnedTriggerName) return;
+
+	return workflow.nodes.find(
+		({ type, name }) => isTrigger(type) && name === firstPinnedTriggerName,
+	);
+}
```
Comment (alexgrozav) [inline packages/cli/src/Server.ts:3116]: We've got code duplication here. Can we import `findPinnedTrigger` from core instead? (also, make it a standalone helper function in core to have it importable)

File: packages/cli/src/Server.ts [hunk 2]
```diff
@@ -1085,12 +1087,15 @@ class App {
 
 					const sessionId = GenericHelpers.getSessionId(req);
 
+					const pinnedTrigger = findPinnedTrigger(workflowData, pinData);
```
Comment (alexgrozav) [inline packages/cli/src/Server.ts:1090]: How do we handle the scenario in which we have multiple pinned triggers? WIll only one get executed?

**Review by alexgrozav — CHANGES_REQUESTED** [review]

**Comment by ivov** [comment]
@alexgrozav This was not sent for review.

**Review by alexgrozav — APPROVED** [review]
Works as expected! Great work.

**Pull request merged**